File: app/src/main/java/temple/edu/yelp_randomizer/View/DetailsView.java
package temple.edu.yelp_randomizer.View;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import com.squareup.picasso.Picasso;
import temple.edu.yelp_randomizer.R;

import java.util.HashMap;

public class DetailsView extends ConstraintLayout {
    private ImageView firstImageView;
    private ImageView secondImageView;
    private ImageView thirdImageView;
    private TextView sundayTextView;
    private TextView mondayTextView;
    private TextView tuesdayTextView;
    private TextView wednesdayTextView;
    private TextView thursdayTextView;
    private TextView fridayTextView;
    private TextView saturdayTextView;

    private Context context;

    public DetailsView( Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
        LayoutInflater inflater = LayoutInflater.from(context);
        inflater.inflate(R.layout.details_view, this, true);

        this.firstImageView = findViewById(R.id._firstDetailsImageView);
        this.secondImageView = findViewById(R.id._secondDetailsImageView);
        this.thirdImageView = findViewById(R.id._thirdDetailsImageView);

        this.sundayTextView = findViewById(R.id._sundayTextView);
        this.mondayTextView = findViewById(R.id._mondayTextView);
        this.tuesdayTextView = findViewById(R.id._tuesdayTextView);
        this.wednesdayTextView = findViewById(R.id._wednesdayTextView);
        this.thursdayTextView = findViewById(R.id._thursdayTextView);
        this.fridayTextView = findViewById(R.id._fridayTextView);
        this.saturdayTextView = findViewById(R.id._saturdayTextView);
    }

    public void setFirstImageView(String imageURL) {
        if (imageURL.length() > 1) {
            Picasso.with(context).load(imageURL).into(firstImageView);
        }
    }

    public void setSecondImageView(String imageURL) {
        if (imageURL.length() > 1) {
            Picasso.with(context).load(imageURL).into(secondImageView);
        }
    }

    public void setThirdImageView(String imageURL) {
        if (imageURL.length() > 1) {
            Picasso.with(context).load(imageURL).into(thirdImageView);
        }
    }

    public void setSundayTextView(String text) {
        this.sundayTextView.setText(text);
    }

    public void setMondayTextView(String text) {
        this.mondayTextView.setText(text);
    }

    public void setTuesdayTextView(String text) {
        this.tuesdayTextView.setText(text);
    }

    public void setWednesdayTextView(String text) {
        this.wednesdayTextView.setText(text);
    }

    public void setThursdayTextView(String text) {
        this.thursdayTextView.setText(text);
    }

    public void setFridayTextView(String text) {
        this.fridayTextView.setText(text);
    }

    public void setSaturdayTextView(String text) {
        this.saturdayTextView.setText(text);
    }


}

File: app/src/main/java/temple/edu/random/View/DetailsView.kt
package temple.edu.random.View

import android.content.Context
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import android.widget.TextView
import com.squareup.picasso.Picasso
import android.view.LayoutInflater
import android.widget.ImageView
import temple.edu.random.R

class DetailsView( context: Context, attrs: AttributeSet?) : ConstraintLayout(context, attrs) {
    private val firstImageView: ImageView
    private val secondImageView: ImageView
    private val thirdImageView: ImageView
    private val sundayTextView: TextView
    private val mondayTextView: TextView
    private val tuesdayTextView: TextView
    private val wednesdayTextView: TextView
    private val thursdayTextView: TextView
    private val fridayTextView: TextView
    private val saturdayTextView: TextView
    fun setFirstImageView(imageURL: String) {
        if (imageURL.length > 1) {
            Picasso.with(context).load(imageURL).into(firstImageView)
        }
    }

    fun setSecondImageView(imageURL: String) {
        if (imageURL.length > 1) {
            Picasso.with(context).load(imageURL).into(secondImageView)
        }
    }

    fun setThirdImageView(imageURL: String) {
        if (imageURL.length > 1) {
            Picasso.with(context).load(imageURL).into(thirdImageView)
        }
    }

    fun setSundayTextView(text: String?) {
        sundayTextView.text = text
    }

    fun setMondayTextView(text: String?) {
        mondayTextView.text = text
    }

    fun setTuesdayTextView(text: String?) {
        tuesdayTextView.text = text
    }

    fun setWednesdayTextView(text: String?) {
        wednesdayTextView.text = text
    }

    fun setThursdayTextView(text: String?) {
        thursdayTextView.text = text
    }

    fun setFridayTextView(text: String?) {
        fridayTextView.text = text
    }

    fun setSaturdayTextView(text: String?) {
        saturdayTextView.text = text
    }

    init {
        val inflater = LayoutInflater.from(context)
        inflater.inflate(R.layout.details_view, this, true)
        firstImageView = findViewById(R.id._firstDetailsImageView)
        secondImageView = findViewById(R.id._secondDetailsImageView)
        thirdImageView = findViewById(R.id._thirdDetailsImageView)
        sundayTextView = findViewById(R.id._sundayTextView)
        mondayTextView = findViewById(R.id._mondayTextView)
        tuesdayTextView = findViewById(R.id._tuesdayTextView)
        wednesdayTextView = findViewById(R.id._wednesdayTextView)
        thursdayTextView = findViewById(R.id._thursdayTextView)
        fridayTextView = findViewById(R.id._fridayTextView)
        saturdayTextView = findViewById(R.id._saturdayTextView)
    }
}